webapp-admin/src/components/SigningLog.js

```diff
@@ -45,33 +53,48 @@ class SigningLog extends Component {
   }
 
   handleItemClick = (index, key) => {
-    this.setState({startRow: 0, endRow: PAGINATION_SIZE});
-    this.props.onItemClick(index, key)
+    var items;
+    if (key === 'models') {
+      items = this.state.filterModels;
+      items[index].selected = !items[index].selected;
+      var filter = items.filter(i => i.selected === true).map(i => i.name).join(',')
+      this.getSigningLogs(this.state.authorityID, 0, filter, this.state.query)
+      this.setState({filterModels: items, filterString: filter});
+    }
   }
 
-  handleRecordsForPage = (startRow, endRow) => {
-    this.setState({startRow: startRow, endRow: endRow});
+  getSigningLogs = (authorityID, offset, filter, serialnumber) => {
```

Comment:
If `authorityID` is always `this.state.authorityID` you can change these args to:
`getSigningLogs = (offset, filter, serialnumber)`
Then get the state within the function:
`var authorityID = this.state.authorityID;`
That way you don't need to pass the state from every call to this.

Reply:
fixed